### DISCUSSION TITLE
Now I always have this warning: "Cannot access configuration for lua_ls. Ensure this server is listed in `server_configurations.md` or added as a custom server."

### DISCUSSION THREAD

**Discussion by neuromaancer** [LazyVim/LazyVim#1721] [General]
Now I always have the warning: "Cannot access configuration for lua_ls. Ensure this server is listed in `server_configurations.md` or added as a custom server.". When I open the `LazyVim` configuration, I am not sure if someone has the same issue.

**Comment by dpetka2001** [comment]
Did you change something in your configuration about `lspconfig`? Out of the box it should just work. Does `mason` list `lua_ls` as installed? By default it's installed by LazyVim.

**Reply by Darkoknight2009** [reply]
yes i believe this right i tried and it worked for me

**Comment by neuromaancer** [comment]
@dpetka2001 

> Did you change something in your configuration about `lspconfig`? Out of the box it should just work. Does `mason` list `lua_ls` as installed? By default it's installed by LazyVim.

I changed a bit but not too much, my `lspconfig.lua`:
```lua
return {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
        ---@type lspconfig.options
        servers = {
            pyright = { autoformat = true },
            ruff_lsp = {
                autostart = true,
            },
            rust_analyzer = { autoformat = true },
            lua_ls = {
                -- mason = false, -- set to false if you don't want this server to be installed with mason
                -- Use this to add any additional keymaps
                -- for specific lsp servers
                ---@type LazyKeysSpec[]
                -- keys = {},
                settings = {
                    Lua = {
                        workspace = {
                            checkThirdParty = false,
                        },
                        completion = {
                            callSnippet = "Replace",
                        },
                    },
                },
            },
        },
    },
}
```
The section of lua_ls is completely same as `LazyVim`. I have installed `lua-language-server` in mason for sure.

**Reply by dpetka2001** [reply]
This looks correct to me. I don't know about your configuration, but maybe you could test with a new Lazyvim starter template and see that it works and gradually add the things you know, so that you find out what broke your configuration.

**Reply by neuromaancer** [reply]
Yeah, it is very painful. What do you think could be the reason? I think it might be the `nvim-lint.lua`, but I don't think i made something wrong in my file:
```lua
﻿return {
    "mfussenegger/nvim-lint",
    opts = {
        events = { "BufWritePost", "BufReadPost", "InsertLeave" },
        linters_by_ft = {
            lua = { "luacheck" },
            python = { "ruff" },
            yaml = { "yamllint" },
            markdown = { "markdownlint" },
            latex = { "chktex" },
        },
    },
}
```
What you think?

**Comment by folke** [comment]
The error you are seeing is from lspconfig and is because this fails:

```lua
require('lspconfig.server_configurations.lua_ls')
```

Try running that in Neovim with `:lua require...` to see the actual error.

**Reply by neuromaancer** [reply]
Thanks! I found this:

```
   Error  18:28:33 msg_show.lua_error   require('lspconfig.server_configurations.lua_ls') E5108: Error executing lua [string ":lua"]:1: module 'lspconfig.server_configurations.lua_ls' not found:
	no field package.preload['lspconfig.server_configurations.lua_ls']
cache_loader: module lspconfig.server_configurations.lua_ls not found
cache_loader_lib: module lspconfig.server_configurations.lua_ls not found
	no file './lspconfig/server_configurations/lua_ls.lua'
	no file '/opt/homebrew/share/luajit-2.1/lspconfig/server_configurations/lua_ls.lua'
	no file '/usr/local/share/lua/5.1/lspconfig/server_configurations/lua_ls.lua'
	no file '/usr/local/share/lua/5.1/lspconfig/server_configurations/lua_ls/init.lua'
	no file '/opt/homebrew/share/lua/5.1/lspconfig/server_configurations/lua_ls.lua'
	no file '/opt/homebrew/share/lua/5.1/lspconfig/server_configurations/lua_ls/init.lua'
	no file './lspconfig/server_configurations/lua_ls.so'
	no file '/usr/local/lib/lua/5.1/lspconfig/server_configurations/lua_ls.so'
	no file '/opt/homebrew/lib/lua/5.1/lspconfig/server_configurations/lua_ls.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file './lspconfig.so'
	no file '/usr/local/lib/lua/5.1/lspconfig.so'
	no file '/opt/homebrew/lib/lua/5.1/lspconfig.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
	[C]: in function 'require'
	[string ":lua"]:1: in main chunk
```

This is so strange, I run `:checkhealth`. In `neoconf.nvim`, it said I didn't install `lspconfig lua_ls`. Actually I have already installed via `Mason` see the screenshot below:
![image](https://github.com/LazyVim/LazyVim/assets/28112871/df804423-ee41-489a-b573-b5e27f22411b)

**Reply by folke** [reply]
This means you dont have `lspconfig` installed. Has nothing to do with mason. Did you disable it for some reason?

**Reply by neuromaancer** [reply]
I have a very minimal setup, I don't think I have disabled it. The list of plugins i have:
```bash
├── LICENSE
├── init.lua
├── lazy-lock.json
├── lazyvim.json
├── lua
│   ├── config
│   │   ├── autocmds.lua
│   │   ├── keymaps.lua
│   │   ├── lazy.lua
│   │   └── options.lua
│   └── plugins
│       ├── cmp.lua
│       ├── colorizer.lua
│       ├── conform.lua
│       ├── cursorword.lua
│       ├── header.lua
│       ├── indentline.lua
│       ├── lint.lua
│       ├── lspconfig.lua
│       ├── mason.lua
│       ├── neogen.lua
│       ├── notify.lua
│       ├── smartcolumn.lua
│       ├── telescope.lua
│       └── themer.lua
└── stylua.toml
```
I have checked again, there is no opt tha disabled the `lspconfig`. 

lazyvim.json:

```json
{
  "extras": [
    "lazyvim.plugins.extras.coding.copilot",
    "lazyvim.plugins.extras.coding.yanky",
    "lazyvim.plugins.extras.dap.core",
    "lazyvim.plugins.extras.editor.navic",
    "lazyvim.plugins.extras.editor.symbols-outline",
    "lazyvim.plugins.extras.formatting.black",
    "lazyvim.plugins.extras.formatting.prettier",
    "lazyvim.plugins.extras.lang.json",
    "lazyvim.plugins.extras.lang.markdown",
    "lazyvim.plugins.extras.lang.python",
    "lazyvim.plugins.extras.lang.python-semshi",
    "lazyvim.plugins.extras.lang.rust",
    "lazyvim.plugins.extras.lang.tex",
    "lazyvim.plugins.extras.lang.yaml",
    "lazyvim.plugins.extras.test.core",
    "lazyvim.plugins.extras.ui.edgy",
    "lazyvim.plugins.extras.ui.mini-animate",
    "lazyvim.plugins.extras.util.mini-hipatterns"
  ],
  "news": {
    "NEWS.md": "2123"
  },
  "version": 2
}
```
My system is MacOS 14.0.

NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1696795921

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.9.4/share/nvim"
 
I just put my config here: https://github.com/neuromaancer/nvim_config

**Reply by folke** [reply]
Does `:Lazy check` show any errors?
Try deleting `~/.local/share/nvim/lazy/nvim-lspconfig` and restart Neovim

**Reply by neuromaancer** [reply]
Nothing happened with `:Lazy check` and I just deleted `nvim-lspconfig` in share, the same error.

**Reply by folke** [reply]
remove `version = "*"` from your `lazy.lua.

I'll actually remove that comment from the starter since that breaks peoples configs more often than not

**Reply by neuromaancer** [reply]
Amazing! Fixed it! But still don't know why there is an error like that. Anyway! Thank you so much!
